base/sort.jl

```diff
@@ -789,16 +789,55 @@ partialsortperm(v::AbstractVector, k::Union{Integer,OrdinalRange}; kwargs...) =
 """
     partialsortperm!(ix, v, k; by=<transform>, lt=<comparison>, rev=false, initialized=false)
 
-Like [`partialsortperm`](@ref), but accepts a preallocated index vector `ix`. If `initialized` is `false`
-(the default), `ix` is initialized to contain the values `1:length(ix)`.
-"""
+Like [`partialsortperm`](@ref), but accepts a preallocated index vector `ix` the same size as
+`v`, which is used to store (a permutation of) the indices of `v`.
+
+If the index vector `ix` is initialized with the indices of `v` (or a permutation thereof), `initialized` should be set to
+`true`.
+
+If `initialized` is `false` (the default), then `ix` is initialized to contain the indices of `v`.
+
+If `initialized` is `true`, but `ix` does not contain (a permutation of) the indices of `v`, the behavior of
+`partialsortperm!` is undefined.
+
+(Typically, the indices of `v` will be `1:length(v)`, although if `v` has an alternative array type
+with non-one-based indices, such as an `OffsetArray`, `ix` must also be an `OffsetArray` with the same
+indices, and must contain as values (a permutation of) these same indices.)
+
+Upon return, `ix` is guaranteed to have the indices `k` in their sorted positions, such that
```

Comment:
Would 

```jl
partialsortperm!(ix, v, k)
v[ix[k]] == partialsort!(v, k)
```

be more clear here?

Reply:
My point was also that the example is talking about guarantees of the state of `ix` on return but then doesn't use that directly, but instead uses the return value which is not identical.